src/interchange/src/avro.rs

```diff
@@ -2059,54 +2059,72 @@ impl DebeziumDecodeState {
     writer_schemas: Option<SchemaCache>,
+    confluent_wire_format: bool,
 }
 
 impl ConfluentAvroResolver {
-    pub fn new(reader_schema: &str, config: Option<ccsr::ClientConfig>) -> anyhow::Result<Self> {
+    pub fn new(
+        reader_schema: &str,
+        config: Option<ccsr::ClientConfig>,
+        confluent_wire_format: bool,
+    ) -> anyhow::Result<Self> {
         let reader_schema = parse_schema(reader_schema)?;
         let writer_schemas =
             config.map(|sr| SchemaCache::new(sr, reader_schema.fingerprint::<Sha256>()));
         Ok(Self {
             reader_schema,
             writer_schemas,
+            confluent_wire_format,
         })
     }
 
     pub async fn resolve<'a, 'b>(
         &'a mut self,
         mut bytes: &'b [u8],
     ) -> anyhow::Result<(&'b [u8], &'a Schema)> {
-        // The first byte is a magic byte (0) that indicates the Confluent
-        // serialization format version, and the next four bytes are a big
-        // endian 32-bit schema ID.
-        //
-        // https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
-        if bytes.len() < 5 {
-            bail!(
-                "Confluent-style avro datum is too few bytes: expected at least 5 bytes, got {}",
-                bytes.len()
-            );
-        }
-        let magic = bytes[0];
-        let schema_id = BigEndian::read_i32(&bytes[1..5]);
-        bytes = &bytes[5..];
-
-        if magic != 0 {
-            bail!(
-                "wrong Confluent-style avro serialization magic: expected 0, got {}",
-                magic
-            );
-        }
+        let mut extract_schema_id = || {
+            // The first byte is a magic byte (0) that indicates the Confluent
+            // serialization format version, and the next four bytes are a big
+            // endian 32-bit schema ID.
+            //
+            // https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+            if bytes.len() < 5 {
+                bail!(
+                        "Confluent-style avro datum is too few bytes: expected at least 5 bytes, got {}",
+                        bytes.len()
+                    );
+            }
+            let magic = bytes[0];
+            let schema_id = BigEndian::read_i32(&bytes[1..5]);
+            bytes = &bytes[5..];
 
+            if magic != 0 {
+                bail!(
+                    "wrong Confluent-style avro serialization magic: expected 0, got {}",
+                    magic
+                );
+            }
+            Ok(schema_id)
+        };
         let resolved_schema = match &mut self.writer_schemas {
-            Some(cache) => cache
-                .get(schema_id, &self.reader_schema)
-                .await
-                .with_context(|| format!("Failed to fetch schema with ID {}", schema_id))?,
+            Some(cache) => {
+                let schema_id = extract_schema_id()?;
```

Comment:
Can we assert that `confluent_wire_format` is true here (and explain why it must be) ?

Reply:
I'll add it as a debug assert, technically it's just ignored in this branch because we have to use it to resolve with the writer schema, and I don't think that anything can go wrong with processing anything if it's set incorrectly.